base/src/main/java/com/smartdevicelink/managers/ManagerUtility.java

```diff
@@ -0,0 +1,86 @@
+         * @return true if name exist in WindowCapability else false
+         */
+        public static boolean hasImageFieldOfName(WindowCapability windowCapability, ImageFieldName name) {
+            if (windowCapability == null) {
+                return false;
+            }
+            if (windowCapability.getImageFields() != null) {
+                for (ImageField field : windowCapability.getImageFields()) {
+                    if (field != null && field.getName() != null && field.getName().equals(name)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        /**
+         * Check to see if WindowCapability has a textField of a given name.
+         *
+         * @param windowCapability WindowCapability representing the capabilities of the desired window
+         * @param name TextFieldName representing a name of a given text field that would be stored in WindowCapability
+         * @return true if name exist in WindowCapability else false
+         */
+        public static boolean hasTextFieldOfName(WindowCapability windowCapability, TextFieldName name) {
+            if (windowCapability == null) {
+                return false;
+            }
+            if (windowCapability.getTextFields() != null) {
+                for (TextField field : windowCapability.getTextFields()) {
+                    if (field != null && field.getName() != null && field.getName().equals(name)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        /**
+         * Method to get number of textFields allowed to be set according to WindowCapability
+         *
+         * @param windowCapability WindowCapability representing the capabilities of the desired window
+         * @return linesFound Number of textFields found in WindowCapability
+         */
+        public static int getMaxNumberOfMainFieldLines(WindowCapability windowCapability) {
+            int highestFound = 0;
+            TextFieldName name;
+            if (windowCapability != null && windowCapability.getTextFields() != null) {
+                for (TextField field : windowCapability.getTextFields()) {
+                    if (field.getName() != null) {
+                        name = field.getName();
+                        if (name == TextFieldName.mainField1 || name == TextFieldName.mainField2 || name == TextFieldName.mainField3 || name == TextFieldName.mainField4) {
+                            int fieldNumber = Integer.parseInt(name.toString().substring(name.toString().length() - 1));
+                            highestFound = Math.max(highestFound, fieldNumber);
+                            if (highestFound == 4) {
+                                break;
+                            }
+                        }
```

Comment:
```suggestion

                        int fieldNumber =0;
                        switch (textField.getName()){
                            case mainField1:
                                fieldNumber = 1;
                                break;
                            case mainField2:
                                fieldNumber = 2;
                                break;
                            case mainField3:
                                fieldNumber = 3;
                                break;
                            case mainField4:
                                fieldNumber = 4;
                                break;
                        }
                        if(fieldNumber > 0){
                            highestFound = Math.max(highestFound, fieldNumber);
                            if (highestFound == 4) {
                                break;
                            }
                        }
```

Reply:
While the current solution is more adaptable/clever, the reality is that there are only 4 specific cases that we care about and therefore should avoid string manipulation. I suggest not taking this suggestions from github, but making the changes in Android Studio.

Reply:
I agree that the suggested code should be better in performance. I will update the method.